app/controllers/devices_controller.rb

```diff
@@ -12,6 +12,10 @@ def new
     @device = Device.new
   end
 
+  def edit
+    @device = Device.find(params[:id])
```

Comment:
move the device finding to a before_action in order to have cleaner and DRY code